addons/account/models/account_move.py

```diff
@@ -3343,7 +3342,7 @@ def field_will_change(line, field_name):
                 return line[field_name].id != vals[field_name]
             if field.type in ('one2many', 'many2many'):
                 current_ids = set(line[field_name].ids)
-                after_write_ids = set(r['id'] for r in line.resolve_2many_commands(field_name, vals[field_name], fields=['id']))
+                after_write_ids = set(getattr(line.new({field_name: vals[field_name]}), field_name).ids)
```

Comment:
Avoid using `getattr` to get a field by its name, use the `[]` operator instead:
```suggestion
                after_write_ids = set(line.new({field_name: vals[field_name]})[field_name].ids)
```